mirrord/config/src/external_proxy.rs

```diff
@@ -0,0 +1,110 @@
+use std::{
+    net::SocketAddr,
+    path::{Path, PathBuf},
+};
+
+use mirrord_config_derive::MirrordConfig;
+use schemars::JsonSchema;
+use serde::Serialize;
+
+use crate::config::source::MirrordConfigSource;
+
+/// Configuration for the external proxy mirrord spawns for mirrord container support, this proxy is
+/// used to allow intproxy running in sidecar to connect to agent
+///
+/// This is seldom used, but if you get `ConnectionRefused` errors, you might
+/// want to increase the timeouts a bit.
+///
+/// ```json
+/// {
+///   "external_proxy": {
+///     "start_idle_timeout": 30,
+///     "idle_timeout": 5
+///   }
+/// }
+/// ```
+#[derive(MirrordConfig, Clone, Debug, Serialize)]
+#[config(map_to = "ExternalProxyFileConfig", derive = "JsonSchema")]
+#[cfg_attr(test, config(derive = "PartialEq"))]
+pub struct ExternalProxyConfig {
+    /// ```json
+    /// {
+    ///   "exyernal_proxy": {
+    ///     "connect_tcp": "10.10.0.100:7777"
+    ///   }
+    /// }
+    /// ```
+    #[config(env = "MIRRORD_EXTERNAL_CONNECT_TCP")]
+    pub connect_tcp: Option<SocketAddr>,
```

Comment:
I think 99% of the configuratoins are internal and should never be used by user, so either hide it or use another means to load this config..

Reply:
I've hidden most from output of medschool